sys/kern/sys_process.c

```diff
@@ -1033,43 +1033,35 @@ kern_ptrace(struct thread *td, int req, pid_t pid, void * __capability addr, int
 		case PT_TO_SCE:
 		case PT_TO_SCX:
 		case PT_SYSCALL:
-			if ((__cheri_addr uintptr_t)addr != 1) {
+			if ((__cheri_addr ptraddr_t)addr != 1) {
 				error = ptrace_set_pc(td2,
-				    (u_long)(__cheri_addr uintfptr_t)addr);
+				    (__cheri_addr u_long)addr);
 				if (error)
 					goto out;
 			}
 			switch (req) {
 			case PT_TO_SCE:
 				p->p_ptevents |= PTRACE_SCE;
 				CTR4(KTR_PTRACE,
-		    "PT_TO_SCE: pid %d, events = %#x, PC = %#lx, sig = %d",
-				    p->p_pid, p->p_ptevents,
-				    (u_long)(__cheri_addr uintfptr_t)addr,
-				    data);
+		    "PT_TO_SCE: pid %d, events = %#x, PC = %#lp, sig = %d",
+				    p->p_pid, p->p_ptevents, addr, data);
 				break;
 			case PT_TO_SCX:
 				p->p_ptevents |= PTRACE_SCX;
 				CTR4(KTR_PTRACE,
-		    "PT_TO_SCX: pid %d, events = %#x, PC = %#lx, sig = %d",
-				    p->p_pid, p->p_ptevents,
-				    (u_long)(__cheri_addr uintfptr_t)addr,
-				    data);
+		    "PT_TO_SCX: pid %d, events = %#x, PC = %#lp, sig = %d",
+				    p->p_pid, p->p_ptevents, addr, data);
 				break;
 			case PT_SYSCALL:
 				p->p_ptevents |= PTRACE_SYSCALL;
 				CTR4(KTR_PTRACE,
-		    "PT_SYSCALL: pid %d, events = %#x, PC = %#lx, sig = %d",
-				    p->p_pid, p->p_ptevents,
-				    (u_long)(__cheri_addr uintfptr_t)addr,
-				    data);
+		    "PT_SYSCALL: pid %d, events = %#x, PC = %#lp, sig = %d",
```

Comment:
I don't know that we have '%#lp' support in ktrdump in userland btw.  I also don't recall if we fixed KTR for Cheri to store uintptr_t rather than longs.  ktrdump is likely quite broken on CHERI anyway, or at least you'd have to have purecap ktrdump for a purecap kernel.

Reply:
Ah, and hybrid would be rather sad as it'd capture the pointer to the capability. So, probably best to make this all u_long still.

Reply:
Re-reading the code looks like the correctness for KTR support is dubious for the CHERI kernel. I think these should stay as u_long, for now at least.